pkg/kube/statefulset/statefulset.go

```diff
@@ -78,3 +81,14 @@ func IsReady(sts appsv1.StatefulSet) bool {
 	allReady := replicas == sts.Status.ReadyReplicas
 	return allUpdated && allReady
 }
+
+// HaveEqualSpec accepts a StatefulSet built, and the StatefulSet that existed already
+// in the API server. HaveEqualSpec returns true if all the changes made in the built StatefulSet
```

Comment:
can you check this function's doc? the function does not take into account Sts existing in the api server or being built, only cares about 2 `appsv1.StatefulSet` objects that are being compared. It could read something like: "checks if both arguments have identical `Spec`s."

You could also mention that the compare function will mask existingSts to only consider the attributes in `builtSts` and not every attribute posible. (am I right on this?)